Makefile changes for ONIE

Copyright (C) 2015 Curt Brune <curt@cumulusnetworks.com>

SPDX-License-Identifier:     GPL-2.0

Minor Makefile changes for ONIE build environment.

diff --git a/Make.defaults b/Make.defaults
index 6899773..661b36d 100644
--- a/Make.defaults
+++ b/Make.defaults
@@ -6,6 +6,8 @@ bindir	?= /usr/bin/
 PCDIR	?= $(libdir)/pkgconfig/
 CC	:= $(if $(filter default,$(origin CC)),gcc,$(CC))
 CCLD	:= $(if $(filter undefined,$(origin CCLD)),$(CC),$(CCLD))
+HOST_CC	:= gcc
+HOST_CCLD	:= $(HOST_CC)
 CFLAGS	?= -O0 -g
 
 ARCH = $(shell uname -m)
@@ -17,11 +19,20 @@ cflags	:= $(CFLAGS) \
 	-D_GNU_SOURCE -I${TOPDIR}/src/include/efivar/ \
 	$(if $(filter $(CC),clang),$(clang_cflags),) \
 	$(if $(filter $(CC),gcc),$(gcc_cflags),)
+host_cflags	:= $(HOST_CFLAGS) \
+	-Wall -Wsign-compare -std=gnu11 -fshort-wchar -fPIC \
+	-fvisibility=hidden \
+	-D_GNU_SOURCE -I${TOPDIR}/src/include/efivar/ \
+	$(if $(filter $(HOST_CC),clang),$(clang_cflags),) \
+	$(if $(filter $(HOST_CC),gcc),$(gcc_cflags),)
 clang_ccldflags =
 gcc_ccldflags = -fno-merge-constants
 ccldflags := $(cflags) $(CCLDFLAGS) $(LDFLAGS) \
 	$(if $(filter $(CCLD),clang),$(clang_ccldflags),) \
 	$(if $(filter $(CCLD),gcc),$(gcc_ccldflags),)
+host_ccldflags := $(host_cflags) $(HOST_CCLDFLAGS) $(HOST_LDFLAGS) \
+	$(if $(filter $(HOST_CCLD),clang),$(clang_ccldflags),) \
+	$(if $(filter $(HOST_CCLD),gcc),$(gcc_ccldflags),)
 LIBFLAGS += -shared
 
 SONAME_VERSION := 0
diff --git a/Makefile b/Makefile
index 4b2beb8..1ba3776 100644
--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,12 @@
 TOPDIR = $(shell echo $$PWD)
 
-SUBDIRS := src docs
-VERSION := 0.18
+# ONIE build environment changes
+export CC = $(CROSS_COMPILE)gcc
+export LDFLAGS += $(ONIE_LDFLAGS)
+export CFLAGS  += $(ONIE_CFLAGS)
+
+SUBDIRS := src
+VERSION := 0.18-onie
 
 all : $(SUBDIRS) efivar.spec
 
diff --git a/src/Makefile b/src/Makefile
index 6161399..dbaad8f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -8,7 +8,6 @@ PCTARGETS = efivar.pc efiboot.pc
 BINTARGETS = efivar
 INCTARGETS = include/efivar/efivar-guids.h
 all : $(LIBTARGETS) $(PCTARGETS) $(BINTARGETS) $(INCTARGETS) libefiboot.so
-	@$(MAKE) -C test TOPDIR=$(TOPDIR) SRCDIR=$(SRCDIR)/test $@
 
 EFIVAR_OBJECTS = dp.o dp-acpi.o dp-hw.o dp-media.o dp-message.o \
 	efivarfs.o export.o guid.o guidlist.o guid-symbols.o \
@@ -52,13 +51,13 @@ efivar : efivar.o libefivar.so
 include/efivar/efivar.h : include/efivar/efivar-guids.h
 
 fakeguid.o : guid.c
-	$(CC) $(cflags) -DEFIVAR_BUILD_ENVIRONMENT -c -o $@ $^
+	$(HOST_CC) $(host_cflags) -DEFIVAR_BUILD_ENVIRONMENT -c -o $@ $^
 
 makeguids.o : makeguids.c
-	$(CC) $(cflags) -DEFIVAR_BUILD_ENVIRONMENT -c -o $@ $^
+	$(HOST_CC) $(host_cflags) -DEFIVAR_BUILD_ENVIRONMENT -c -o $@ $^
 
 makeguids : makeguids.o fakeguid.o
-	$(CC) $(cflags) -o $@ $^ -ldl
+	$(HOST_CCLD) $(host_ccldflags) -o $@ $^ -ldl
 
 include/efivar/efivar-guids.h : makeguids guids.txt
 	./makeguids guids.txt guids.bin names.bin guid-symbols.S $@
